pkg/controllers/machine/termination/controller.go

```diff
@@ -0,0 +1,140 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package termination
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"golang.org/x/time/rate"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/client-go/util/workqueue"
+	"knative.dev/pkg/logging"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
+	"github.com/aws/karpenter-core/pkg/cloudprovider"
+	"github.com/aws/karpenter-core/pkg/controllers/machine/termination/terminator"
+	terminatorevents "github.com/aws/karpenter-core/pkg/controllers/machine/termination/terminator/events"
+	"github.com/aws/karpenter-core/pkg/events"
+	corecontroller "github.com/aws/karpenter-core/pkg/operator/controller"
+	machineutil "github.com/aws/karpenter-core/pkg/utils/machine"
+)
+
+var _ corecontroller.FinalizingTypedController[*v1alpha5.Machine] = (*Controller)(nil)
+
+// Controller is a Machine Controller
+type Controller struct {
+	kubeClient    client.Client
+	cloudProvider cloudprovider.CloudProvider
+	recorder      events.Recorder
+	terminator    *terminator.Terminator
+}
+
+// NewController is a constructor for the Machine Controller
+func NewController(kubeClient client.Client, cloudProvider cloudprovider.CloudProvider,
+	terminator *terminator.Terminator, recorder events.Recorder) corecontroller.Controller {
+	return corecontroller.Typed[*v1alpha5.Machine](kubeClient, &Controller{
+		kubeClient:    kubeClient,
+		cloudProvider: cloudProvider,
+		recorder:      recorder,
+		terminator:    terminator,
+	})
+}
+
+func (*Controller) Name() string {
+	return "machine_termination"
+}
+
+func (c *Controller) Reconcile(_ context.Context, _ *v1alpha5.Machine) (reconcile.Result, error) {
+	return reconcile.Result{}, nil
+}
+
+func (c *Controller) Finalize(ctx context.Context, machine *v1alpha5.Machine) (reconcile.Result, error) {
+	stored := machine.DeepCopy()
+	if !controllerutil.ContainsFinalizer(machine, v1alpha5.TerminationFinalizer) {
+		return reconcile.Result{}, nil
+	}
```

Comment:
This initial check should be similar to the node termination controller, right? We should let any other machine controller do the `kubeClient.delete` which then triggers the `DeletionTimestamp` to be set. Any machine that doesn't have a finalizer should be understood as having its finalizer logic already completed. 
```suggestion
	if !machine.DeletionTimestamp.IsZero() {
		return reconcile.Result{}, nil
	}
```